### PR TITLE
Add Broadcom Error logging support

### PR DESCRIPTION
**Pull request by shitalt**

### CODE REVIEW

File: lib/libutils/ext/include/trace.h [hunk 1]
```diff
@@ -27,6 +27,7 @@ void trace_ext_puts(const char *str);
 int trace_ext_get_thread_id(void);
 void trace_set_level(int level);
 int trace_get_level(void);
+void plat_trace_ext_puts(const char *str __unused);
```
Comment (jforissier) [inline lib/libutils/ext/include/trace.h]: No `__unused` needed here

File: core/arch/arm/plat-bcm/platform_config.h [hunk 2]
```diff
@@ -17,6 +17,14 @@
 
 #define GICD_BASE		0x63c00000
 
+/* Use default AP error logging address */
+#ifndef CFG_BCM_ELOG_AP_UART_LOG_BASE
+#define CFG_BCM_ELOG_AP_UART_LOG_BASE	0x8f110000
+#endif
+
+#define BCM_ELOG_OPTEE_BASE    CFG_BCM_ELOG_AP_UART_LOG_BASE
+#define BCM_ELOG_OPTEE_SIZE    0x10000
```
Comment (jforissier) [inline core/arch/arm/plat-bcm/platform_config.h]: For simplicity: base and size should either be hard-coded here (without `CFG_`), or if they need to be made configurable at compile-time, they should be defined in `conf.mk` as `CFG_BCM_ELOG_OPTEE_BASE ?= 0x8f110000` and `CFG_BCM_ELOG_OPTEE_SIZE ?= 0x10000`. In this case you may use the `CFG_` directly in the code.

File: core/arch/arm/plat-bcm/bcm_elog.c [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+static struct bcm_elog global_elog;
+
+void bcm_elog_putchar(char ch)
+{
+	struct bcm_elog *elog = &global_elog;
+	uint32_t offset = 0, len = 0;
+	vaddr_t base = 0;
+
+	base = (vaddr_t)io_pa_or_va(&elog->base);
```
Comment (jforissier) [inline core/arch/arm/plat-bcm/bcm_elog.c]: Cast not needed

File: lib/libutils/ext/include/trace.h (hunk 1 above)
Comment (shitalt) [reply lib/libutils/ext/include/trace.h]: Done

File: core/arch/arm/plat-bcm/platform_config.h (hunk 2 above)
Comment (shitalt) [reply core/arch/arm/plat-bcm/platform_config.h]: will move config option `CFG_BCM_ELOG_AP_UART_LOG_BASE` to conf.mk and use it directly in code.

File: core/arch/arm/plat-bcm/bcm_elog.c (hunk 3 above)
Comment (shitalt) [reply core/arch/arm/plat-bcm/bcm_elog.c]: Removed here and at other places as well

**Comment by shitalt** [comment]
Hi @jforissier 

> For "plat-bcm: Add Broadcom platform specific err log support":
> 
> `Acked-by: Jerome Forissier <jerome@forissier.org>`

Ack is only for the "plat-bcm: Add Broadcom platform specific err log support" commit or it includes both commits in this series ?

**Comment by jforissier** [comment]
For the first commit I have given a `Reviewed-by` already.

**Comment by shitalt** [comment]
Tags applied

**Pull request merged**